contracts/amm/MinterAmm.sol

```diff
@@ -649,8 +649,8 @@ contract MinterAmm is InitializeableAmm, OwnableUpgradeSafe, Proxiable {
                 uint256 unclaimedPayment = paymentToken
                     .balanceOf(address(optionMarket))
                     .mul(wTokenBalance)
-                    .div(wTokenSupply)
                     .mul(1e18)
+                    .div(wTokenSupply)
```

Comment:
Is there a chance that this can lead to an overflow? unit256 max value is roughly 1e77. If all tokens have 18 decimals then 18 + 18 + 18 = 54 in a mild case where all tokens have single-digit amounts. If we go to higher amounts it can get dangerously close to 77. I think this sequence of operations was decided here on purpose to prevent overflow, which can have much worse consequences than potential precision loss.

Reply:
hmm good point, So the combined decimals of the payment token and wToken balance would need to exceed 22 for this to overflow, which would cause the transaction to fail and would mean any callers of getTotalPoolValue (which includes the `provideCapital` function would be uncallable until `claimAllExpiredTokens` is called.

I disagree that this is a worse consequence than the existing bug. If this state were reached, it would halt the system, but it could easily be unstuck by anyone calling `claimAllExpiredTokens`. I think this state is so rare, and the solution apparent, that we should still go forward with the change. If we keep the existing code, then we have a very subtle bug: the `getTotalPoolValue` will not include the unclaimed paymentToken in its calculation of total pool value, and so any code that relies on `getTotalPoolValue`'s calculated value matching up with the true value of the pool will also not work correctly. This bug is too subtle for small minds such as ours to reason through concretely, and I would rather get rid of this bug in favor of an explicit and easily fixable bug, even though it would be immediately jarring for the protocol's users when triggered.

Reply:
Ok, I agree with this. I missed the point that currently if payment token balance is 0 we're disregarding any unclaimed payment token due to 0-rounding